prototype_experiment.py

```diff
@@ -0,0 +1,279 @@
+datalist = {"imagenet": imagenet, "coco": coco_seg}
+
+#Knob #1.5
+#The selection of the stitching parents from the pretrained models for the given dataset
+imagenet_models = {"resnet50": get_pretrained_model("resnet18"), "resnet101": get_pretrained_model("resnet34"), "ViT": get_pretrained_model("vit_b_32")}
+coco_models = {}
+modellist = {"imagenet": imagenet_models, "coco": coco_models}
+
+#Knob 2
+#The selection of the stitching layer
+stitching_families = {"conv1x1": Conv2d}
+
+#Knob 3 
+#The selection of the loss function
+loss_functions = {"class_labels": 0, "soft_labels": 1}
+
+#Set torch to use cuda GPUs
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+#Dataset root directory, 
+#sShoulnd't it just be datasets? There is no data folder in the src directory?
+#what is number of workers? 
+data_module = datalist["imagenet"]
+
+#prepare and setup data
+data_module.prepare_data()
+data_module.setup("test")
+
+#Get pretrained models that will be stitched together
+modelA = modellist["imagenet"]["resnet50"]
+modelB = modellist["imagenet"]["resnet101"]
+
+#Get the graph trace of our pretrained models 
+traceA = symbolic_trace(modelA)
+traceB = symbolic_trace(modelB)
+modelA = gmp.GraphModulePlus(root = traceA, graph = traceA.graph, class_name = "HK-Arson")
+modelB = gmp.GraphModulePlus(root = traceB, graph = traceB.graph, class_name = "HK-Butcher")
+
+
+
+
+# single conv operations
+#Why do we need to do this? 
+modelA = modelA.squash_all_conv_batchnorm_pairs()
+modelB = modelB.squash_all_conv_batchnorm_pairs()
+
+#get split points between layers, could be much more efficient
+layersA = []
+layersB = []
+
+for node in modelA.graph.nodes: #All the stitch points for modelA
+    if (node.name.count("add") > 0):
+        layersA.append(node.name)
+
+for node in modelB.graph.nodes: #all the stitch points for modelB
+    if (node.name.count("add") > 0):
+        layersB.append(node.name)
+
+layersA = [layersA[2]]
+layersB = [layersB[4]]
+
+# Set up models on the CUDA gpus and put them in eval mode
+modelA = modelA.to(device).eval()
+modelB = modelB.to(device).eval()
+
+def display_model_graph(graph, file, dpi=200):
+    image = graph.to_dot().create_png(prog="dot")
+    with io.BytesIO(image) as f:
+        image = mpimg.imread(f)
+    plt.figure(figsize=(image.shape[1] / dpi, image.shape[0] / dpi), dpi=dpi)
+    plt.imshow(image)
+    plt.axis("off")
+    plt.savefig(file)
+    plt.show()
+
+#display_model_graph(modelA, "temp.png")
+
+#create stitching model, train the stitching layer, record loss value, repeat for all different possible connections 
+#how can I more easily iterate across layers? Across nodes? across different layer sections? How to pull out adds?
+#how to stitch between different sections? 
+#modelA.graph.print_tabular()
+
+
+score = {"name": [], "value": [], "ParentA":  [], "ParentB": []}
+for layerA in layersA:
```

Comment:
aesthetic preference: loop over layers is something I'd do in bash so that one call of the python script = one model trained.

related comment: expect things to fail. Maybe the server gets restarted after doing 1/2 of the analyses. Good practice to (1) check if a model is already trained and (2) quit early (or `continue` the loop if so)